python/tk_houdini/bootstrap.py

```diff
@@ -12,68 +12,274 @@
 import shutil
 import sys
 import tempfile
+import traceback
 
+################################################################################
+# constants defining environment variables used during bootstrap
 
-g_temp_env = 'TK_HOUDINI_TMP'
+# Name of the env variable that stores the temp directory where .xml files are
+# written to defined menus, shelves and python panels for the SG integration.
+g_temp_env = "TK_HOUDINI_TMP"
 
+# Name of the env variable that stores the serialized context used durring
+# classit toolkit bootstrap
+g_sgtk_context_env = "TANK_CONTEXT"
+
+# Name of the env variable that stores the name the engine instance name for
+# classic toolkit bootstrap
+g_sgtk_engine_env = "TANK_ENGINE"
+
+# Name of the file to open after bootstrap
+g_sgtk_file_to_open_env = "TANK_FILE_TO_OPEN"
+
+################################################################################
+# methods for bootstrapping toolkit within houdini
 
 def bootstrap(tank, context):
+    """
+    Interface for older versions of tk-multi-launchapp.
+
+    This is deprecated and now replaced with the ``startup.py`` file and
+    ``SoftwareLauncher`` interface.
+
+    Prepares the environment for a tk-houdini bootstrap. This method is
+    called directly from the tk-multi-launchapp.
+    """
+
+    # get the necessary environment variable for launch
+    env = get_classic_startup_env()
+
+    # update the environment with the classic startup vars
+    os.environ.update(env)
+
+
+def bootstrap_classic():
```

Comment:
This is the logic that used to be duplicated in `startup/python2.X/pythonrc.py` files in the engine root.